### PR TITLE
raylib: 4.5.0 -> 5.0

### PR DESCRIPTION
**Pull request by lucasnethaj**
## Description of changes

Release notes: https://github.com/raysan5/raylib/releases/tag/5.0
Full changelog: https://github.com/raysan5/raylib/blob/master/CHANGELOG#L1-L244

Notable changes (Breaking api):
> [rcore] REMOVED: PLATFORM_RPI (#3232) by michaelfiber
> [rtext] RENAMED: LoadFont*() parameter names for consistency and coherence by raysan5
> [rtextures] REMOVED: GenImageGradientH() and GenImageGradientV() by danemadsen

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by imrying — APPROVED** [review]

**Comment by lucasnethaj** [comment]
OK i just realized why the patch was there in the first place. and realised that upstream forgot to update the Cmake project version again. so I'll update it with the new patch from master

File: pkgs/development/libraries/raylib/default.nix [hunk 1]
```diff
@@ -56,18 +56,18 @@ stdenv.mkDerivation (finalAttrs: {
   passthru.tests = [ raylib-games ];
 
   patches = [
-    # Patch version in CMakeList to 4.5.0
+    # Patch version in CMakeList to 5.0.0
     # Remove this when updating to a new revision
```
Comment (jtbx) [inline pkgs/development/libraries/raylib/default.nix:60]:
```suggestion
    # Patch version in CMakeLists.txt to 5.0.0
    # Change this when updating to a new revision
```

File: pkgs/development/libraries/raylib/default.nix [hunk 2]
```diff
@@ -56,18 +56,18 @@ stdenv.mkDerivation (finalAttrs: {
...
+      hash = "sha256-qsX5AwyQaGoRsbdszOO7tUF9dR+AkEFi4ebNkBVHNEY=";
     })
   ];
 
   # fix libasound.so/libpulse.so not being found
   preFixup = ''
-    ${lib.optionalString alsaSupport "patchelf --add-needed ${alsa-lib}/lib/libasound.so $out/lib/libraylib.so.${finalAttrs.version}"}
-    ${lib.optionalString pulseSupport "patchelf --add-needed ${libpulseaudio}/lib/libpulse.so $out/lib/libraylib.so.${finalAttrs.version}"}
+    ${lib.optionalString alsaSupport "patchelf --add-needed ${alsa-lib}/lib/libasound.so $out/lib/libraylib.so"}
+    ${lib.optionalString pulseSupport "patchelf --add-needed ${libpulseaudio}/lib/libpulse.so $out/lib/libraylib.so"}
```
Comment (jtbx) [inline pkgs/development/libraries/raylib/default.nix:70]: Why is this needed?

**Review by jtbx — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/raylib/default.nix (hunk 2 above)
Comment (lucasnethaj) [reply pkgs/development/libraries/raylib/default.nix:70]: I don't know if it's still needed. I don't for which reason it's was added in the first place.

I removed the `finalAttrs.version` because the git tag did not match with the build version `5,0,0 vs 5,0` so i was not sure i should hard code or just remove it

File: pkgs/development/libraries/raylib/default.nix (hunk 1 above)
Comment (lucasnethaj) [reply pkgs/development/libraries/raylib/default.nix:60]: I don't understand why this is better. it's only necessary to update if they forget to add the build revision again in the release

File: pkgs/development/libraries/raylib/default.nix (hunk 2 above)
Comment (lucasnethaj) [reply pkgs/development/libraries/raylib/default.nix:70]: Ok i just checked and libraries are not even linked anymore

```
> ldd result/lib/libraylib.so.4.5.0 | grep libpulse                                                                          
	/nix/store/ldyiwysirfdxww141v649lslpq9xm0c1-libpulseaudio-16.1/lib/libpulse.so (0x00007f78fa33d000)
	libpulsecommon-16.1.so => /nix/store/ldyiwysirfdxww141v649lslpq9xm0c1-libpulseaudio-16.1/lib/pulseaudio/libpulsecommon-16.1.so (0x00007f78f9b6e000)
> ldd result/lib/libraylib.so.4.5.0 | grep libasound                                                                         
	/nix/store/3ivh62ailk57r674fhk6nx1bg2nhirr7-alsa-lib-1.2.11/lib/libasound.so (0x00007f510608e000)
	
> ldd result/lib/libraylib.so.5.0.0 | grep libasound
> ldd result/lib/libraylib.so.5.0.0 | grep libpulse
```

File: pkgs/development/libraries/raylib/default.nix (hunk 1 above)
Comment (lucasnethaj) [reply pkgs/development/libraries/raylib/default.nix:60]: If the goal is to prevent people from making the same mistake i did. It's better to having something more descriptive like "please check that the output library matches the build version or update this patch"

Comment (jtbx) [reply pkgs/development/libraries/raylib/default.nix:60]: The devs say they're going to update the version number next release: https://github.com/raysan5/raylib/issues/3639

Just keep the comment as-is, the patch should probably be removed next update.

**Comment by jtbx** [comment]
Please squash your commits

File: pkgs/development/libraries/raylib/default.nix (hunk 1 above)
Comment (eclairevoyant) [reply pkgs/development/libraries/raylib/default.nix:60]: > It's better to having something more descriptive like "please check that the output library matches the build version or update this patch"

Then let's go with that. The repo owner doesn't use cmake, so it's liable to happen again.

Comment (lucasnethaj) [reply pkgs/development/libraries/raylib/default.nix]: @eclairevoyant 
> Then let's go with that. The repo owner doesn't use cmake, so it's liable to happen again.
Updated the comment to be more descriptive

File: pkgs/development/libraries/raylib/default.nix [hunk 3]
```diff
@@ -56,20 +56,17 @@ stdenv.mkDerivation (finalAttrs: {
   passthru.tests = [ raylib-games ];
 
   patches = [
-    # Patch version in CMakeList to 4.5.0
-    # Remove this when updating to a new revision
+    # Patch version in CMakeList to 5.0.0
+    # The library author doesn't use cmake
+    # So when updating this package please check that the resulting library extension matches the version
+    # and either remove or update this patch
+    # ie. resulting library should match libraylib.so.${finalAttrs.version}
```
Comment (jtbx) [inline pkgs/development/libraries/raylib/default.nix:63]:
```suggestion
    # Patch version in CMakeLists.txt to 5.0.0
    # The library author doesn't use cmake, so when updating this package please
    # check that the resulting library extension matches the version
    # and remove/update this patch (resulting library name should match
    # libraylib.so.${finalAttrs.version}
```

Comment (lucasnethaj) [reply pkgs/development/libraries/raylib/default.nix]: :+1:

**Review by eclairevoyant — APPROVED** [review]

**Comment by eclairevoyant** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>raylib</li>
    <li>raylib-games</li>
    <li>snekim</li>
  </ul>
</details>

**Review by jtbx — APPROVED** [review]

**Comment by emmanuelrosa** [comment]
Why was the audio patch removed?

Audio is broken on Linux without it.

**Comment by eclairevoyant** [comment]
The audio patchelf hack was only needed because the library version was incorrect, but the patch addresses this. If you're still facing some issue, please file a new GH issue with details and a straightforward way to reproduce the issue.
